storage/watcher.go

```diff
@@ -42,6 +42,9 @@ type WatchStream interface {
 	// Chan returns a chan. All watch response will be sent to the returned chan.
 	Chan() <-chan WatchResponse
 
+	// RequestStatus sends the status of watch in addition to WatchResponse.
```

Comment:
RequestStatus requests the status of the watcher with given ID. The responses will be sent through the WatchRespone Chan attached with this stream to ensure correct ordering.